### PR TITLE
Singlestore & Solr UI support

### PR DESCRIPTION
**Pull request by SayedTahsin**

### CODE REVIEW

File: charts/kubedbcom-solr-editor-options/ui/functions.js [hunk 1]
```diff
@@ -705,9 +727,11 @@ return {
   onCreateAuthSecretChange,
 	getSecrets,
 	disableLimit,
+	disableLimit2,
 	getMachineListForOptions,
 	setResourceLimit,
-	setLimitsCpuOrMem,
+  setResourceLimit2,
```
Comment (nurRiyad) [inline charts/kubedbcom-solr-editor-options/ui/functions.js:733]: fix linting

File: charts/kubedbcom-solr-editor-options/ui/functions.js [hunk 2]
```diff
@@ -434,6 +439,23 @@ function setResourceLimit({ commit, model, getValue, watchDependency }) {
     });
   }
 }
+function setResourceLimit2({ commit, model, getValue, watchDependency },nodeType) {
```
Comment (nurRiyad) [inline charts/kubedbcom-solr-editor-options/ui/functions.js:442]: rename funtion

File: charts/kubedbcom-solr-editor-options/ui/functions.js [hunk 3]
```diff
@@ -434,6 +439,23 @@ function setResourceLimit({ commit, model, getValue, watchDependency }) {
...
+  const modelPathValue = getValue(model, `/spec/topology/${nodeType}/podResources/machine`);
+  watchDependency(`model#/spec/topology/${nodeType}/podResources/machine`);
+  if (modelPathValue && modelPathValue !== "custom") {
+    // to avoiding set value by reference, cpu and memory set separately
+    commit("wizard/model$update", {
+      path: `/spec/topology/${nodeType}/podResources/resources/limits/cpu`,
+      value: machines[modelPathValue].resources.limits.cpu,
```
Comment (nurRiyad) [inline charts/kubedbcom-solr-editor-options/ui/functions.js:449]: use optional parameter

File: charts/kubedbcom-solr-editor-options/ui/functions.js [hunk 4]
```diff
@@ -409,6 +409,11 @@ function disableLimit({ model, getValue, watchDependency }) {
   watchDependency("model#/spec/machine");
   return modelPathValue !== "custom" && !!modelPathValue;
 }
+function disableLimit2({ model, getValue, watchDependency },nodeType) {
```
Comment (nurRiyad) [inline charts/kubedbcom-solr-editor-options/ui/functions.js:412]: rename

File: charts/kubedbcom-singlestore-editor-options/ui/functions.js [hunk 5]
```diff
@@ -417,17 +423,34 @@ function getMachineListForOptions() {
...
+}
+function setResourceLimit2({ commit, model, getValue, watchDependency },nodeType) {
+  const modelPathValue = getValue(model, `/spec/topology/${nodeType}/podResources/machine`);
+  watchDependency(`model#/spec/topology/${nodeType}/podResources/machine`);
   if (modelPathValue && modelPathValue !== "custom") {
     // to avoiding set value by reference, cpu and memory set separately
     commit("wizard/model$update", {
-      path: "/spec/resources/limits/cpu",
+      path: `/spec/topology/${nodeType}/podResources/resources/limits/cpu`,
       value: machines[modelPathValue].resources.limits.cpu,
```
Comment (nurRiyad) [inline charts/kubedbcom-singlestore-editor-options/ui/functions.js:449]: use optional chaining

File: charts/kubedbcom-singlestore-editor-options/ui/create-ui.yaml [hunk 6]
```diff
@@ -155,31 +214,36 @@ steps:
...
-    - computed: setLimitsCpuOrMem|cpu
-      disabled: disableLimit
-      label:
-        text: labels.cpu
-      schema:
-        $ref: schema#/properties/spec/properties/resources/properties/limits/properties/cpu
-      type: input
-    - computed: setLimitsCpuOrMem|memory
-      disabled: disableLimit
+
```
Comment (nurRiyad) [inline charts/kubedbcom-singlestore-editor-options/ui/create-ui.yaml:217]: remove space

**Review by nurRiyad — CHANGES_REQUESTED** [review]

**Review by SamiulSourav — APPROVED** [review]
LGTM

**Review by nurRiyad — APPROVED** [review]
LGTM

**Pull request merged**